delivery_service/PersistantMessageQueue.md

```diff
@@ -0,0 +1,182 @@
+  ```
+  <br/>
+  <b>Kafka</b><br/>
+  You can use the following `docker-compose.yml` and execute `docker-compose up -d` to run the Kafka instance.
+
+  ```
+    version: "3" 
+    services:
+      zookeeper:
+        image: confluentinc/cp-zookeeper:latest
+        networks:
+          - acapy_default
+        environment:
+          ZOOKEEPER_CLIENT_PORT: 2181
+          ZOOKEEPER_TICK_TIME: 2000
+      kafka:
+        image: confluentinc/cp-kafka:latest
+        networks:
+          - acapy_default
+        depends_on:
+          - zookeeper
+        ports:
+          - 29092:29092
+        environment:
+          KAFKA_BROKER_ID: 1
+          KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
+          KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://kafka:9092,PLAINTEXT_HOST://localhost:29092
+          KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: PLAINTEXT:PLAINTEXT,PLAINTEXT_HOST:PLAINTEXT
+          KAFKA_INTER_BROKER_LISTENER_NAME: PLAINTEXT
+          KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
+          KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR: 1
+          KAFKA_TRANSACTION_STATE_LOG_MIN_ISR: 1
+    networks:
+      acapy_default:
+        external: true
+        name: ${DELIVERY_SERVICE_NETWORK_NAME}
+  ```
+- Startup ACA-Py agents with Kafka or Redis backend. Persistent queue specific arguments can be found [here](#startup-arugments).
+  <b>Kafka Inbound and Outbound</b>
+  ```
+  PORTS="5002" ./scripts/run_docker start --admin 0.0.0.0 5002 --admin-insecure-mode --public-invites --auto-ping-connection --auto-accept-invites --auto-accept-requests --auto-respond-credential-offer --auto-respond-credential-request --auto-respond-presentation-proposal --auto-verify-presentation --genesis-url http://host.docker.internal:9000/genesis --seed 00000000000000000000000000000008 --auto-provision --recreate-wallet --wallet-type indy --wallet-name issuer --wallet-key mykey -oq kafka:9092 -oqp acapy -oqc aries_cloudagent.transport.outbound.queue.kafka.KafkaOutboundQueue -iq kafka:9092 -iqp acapy -iqc aries_cloudagent.transport.inbound.queue.kafka.KafkaInboundQueue -e http://host.docker.internal:8002 -iqt http 0.0.0.0 8002
+  ```
+  <b>Redis Inbound and Outbound</b>
+  ```
+  PORTS="5002" ./scripts/run_docker start --admin 0.0.0.0 5002 --admin-insecure-mode --public-invites --auto-ping-connection --auto-accept-invites --auto-accept-requests --auto-respond-credential-offer --auto-respond-credential-request --auto-respond-presentation-proposal --auto-verify-presentation --genesis-url http://host.docker.internal:9000/genesis --seed 00000000000000000000000000000008 --auto-provision --recreate-wallet --wallet-type indy --wallet-name issuer --wallet-key mykey -oq redis://host.docker.internal:6379 -oqp acapy -oqc aries_cloudagent.transport.outbound.queue.redis.RedisOutboundQueue -iq redis://host.docker.internal:6379 -iqp acapy -iqc aries_cloudagent.transport.inbound.queue.redis.RedisInboundQueue -e http://host.docker.internal:8002 -iqt http 0.0.0.0 8002
```

Comment:
Adding newlines to these commands would greatly increase readability; something like:
```sh
./scripts/run_docker start --admin 0.0.0.0 5002 --admin-insecure-mode \
    --public-invites ....
```